.github/workflows/platform-release.yaml

```diff
@@ -0,0 +1,182 @@
+        uses: goreleaser/goreleaser-action@v5
+        with:
+          version: latest
+          distribution: goreleaser
+          args: release --clean --snapshot --config ./.goreleaser.darwin.yaml
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          APPLE_DEVELOPER_ID: ${{ secrets.APPLICATION_IDENTITY}}
+          APPLE_ID: ${{ vars.APPLE_ID }}
+          APPLE_ID_PASSWORD: ${{ secrets.APPLE_ID_PASSWORD}}
+          APPLE_TEAM_ID: ${{ secrets.APPLE_TEAM_ID}}
+      - name: Notarize the macOS binary
+        run: |
+          ./build/scripts/sign ./dist/kitops-darwin-amd64.zip
+          ./build/scripts/sign ./dist/kitops-darwin-x86_64.zip
+      - name: Upload macOS artifacts
+        uses: actions/upload-artifact@v4
+        with:
+            name: dist-macos
+            if-no-files-found: error
+            retention-days: 7
+            path: |
+              ./dist/*.zip
+
+  build-windows:
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+            go-version-file: 'go.mod'
+
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v5
+        with:
+          version: latest
+          distribution: goreleaser
+          args: release --clean --snapshot --config ./.goreleaser.windows.yaml
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Upload Windows artifacts
+        uses: actions/upload-artifact@v4
+        with:
+            name: dist-windows
+            if-no-files-found: error
+            retention-days: 7
+            path: |
+              ./dist/*.zip
+
+  build-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+            go-version-file: 'go.mod'
+
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v5
+        with:
+          version: latest
+          distribution: goreleaser
+          args: release --clean --snapshot --config ./.goreleaser.linux.yaml
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Upload Linux artifacts
+        uses: actions/upload-artifact@v4
+        with:
+            name: dist-linux
+            if-no-files-found: error
+            retention-days: 7
+            path: |
+              ./dist/*.tar.gz
+  # Creates a release with the artifacts from the previous steps.
+  # workflow_dispatch triggered versions will be draft releases.
+  # CLI docs are not updated for workflow_dispatch triggered versions.
+  release:
+    runs-on: ubuntu-latest
+    needs: [build-linux, build-macos, build-windows]
+    steps:
+      - name: Merge built artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: dist
+          pattern: dist-*
+          merge-multiple: true
```

Comment:
I'm not sure how artifacts work -- if we run two releases within a week of each other will they see both artifacts? I suppose the second build would overwrite them?

Reply:
You can only download artifacts that were uploaded during the same workflow run or use the UI to navigate to the workflow run and download. I have also done a shortened retention, the default is 90 days.

Reply:
That makes sense :+1: